src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -195,7 +191,7 @@ private static SequencePosition SeekMultiSegment(ReadOnlySequenceSegment<T> curr
             }
 
             // Hit the end of the segments but didn't reach the count
-            if (currentSegment == null || (currentSegment == endObject && endPosition < offset))
+            if (currentSegment == null || endIndex < offset)
```

Comment:
Why?

Reply:
While at above `while (currentSegment != null && currentSegment != endObject)` ensure that `currentSegment == null || currentSegment == endObject`.

So here `(currentSegment == null || (currentSegment == endObject && endPosition < offset))`
if `currentSegment == null` is false means that `currentSegment == endObject` so we can remove `currentSegment == endObject`.

Reply:
It can be rewrote as `currentSegment != endObject || endPosition < offset` also.